server/handler_test.go

```diff
@@ -188,6 +188,28 @@ func TestHandlerOutput(t *testing.T) {
 		require.Equal(t, sqltypes.Float64, result.Rows[0][0].Type())
 		require.Equal(t, []byte("1"), result.Rows[0][0].ToBytes())
 	})
+
+	t.Run("if() type is correct", func(t *testing.T) {
+		handler.ComInitDB(dummyConn, "test")
+		var result *sqltypes.Result
+		err := handler.ComQuery(dummyConn, "select if(1, 123, 'def')", func(res *sqltypes.Result, more bool) error {
```

Comment:
should be do `(0, "def", 123)` -> `"123`?

Reply:
The old code had `return n.ifTrue.Type()`, which was already Text, so it didn't actually test the new code I added.